recipes/mpir/all/conanfile.py

```diff
@@ -103,6 +103,9 @@ def _configure_autotools(self):
             args.append("--disable-silent-rules")
             args.append("--enable-cxx" if self.options.get_safe("enable_cxx") else "--disable-cxx")
             args.append("--enable-gmpcompat" if self.options.enable_gmpcompat else "--disable-gmpcompat")
+
+            # compiler checks are written for C89 but compilers that default to C99 treat implicit functions as error
+            self._autotools.flags.append("-Wno-implicit-function-declaration")            
```

Comment:
Should we only add this flag for _only_ that particular case?

Reply:
Ideally yes. But how do we detect compilers that come with C99 as default C standard?
Another possibility is to only apply it for the case I found: apple-clang 12
But I guess this flag should be neutral regarding the type of binary that gets produced, so I don't know if it matters to filter or not.